desktop/extensions-sdk/build/set-up/frontend-extension-tutorial.md

```diff
@@ -0,0 +1,309 @@
+  <li><a data-toggle="tab" data-target="#vue-dockerfile" data-group="vue">For Vue</a></li>
+  <li><a data-toggle="tab" data-target="#angular-dockerfile" data-group="angular">For Angular</a></li>
+  <li><a data-toggle="tab" data-target="#svelte-dockerfile" data-group="svelte">For Svelte</a></li>
+</ul>
+
+<div class="tab-content">
+  <div id="react-dockerfile" class="tab-pane fade in active" markdown="1">
+
+```Dockerfile
+FROM --platform=$BUILDPLATFORM node:18.9-alpine3.15 AS client-builder
+WORKDIR /ui
+# cache packages in layer
+COPY ui/package.json /ui/package.json
+COPY ui/package-lock.json /ui/package-lock.json
+RUN --mount=type=cache,target=/usr/src/app/.npm \
+    npm set cache /usr/src/app/.npm && \
+    npm ci
+# install
+COPY ui /ui
+RUN npm run build
+
+FROM alpine
+LABEL org.opencontainers.image.title="My extension" \
+    org.opencontainers.image.description="Your Desktop Extension Description" \
+    org.opencontainers.image.vendor="Docker Inc." \
+    com.docker.desktop.extension.api.version="0.3.0" \
+    com.docker.desktop.extension.icon="https://www.docker.com/wp-content/uploads/2022/03/Moby-logo.png"
+    com.docker.extension.screenshots="" \
+    com.docker.extension.detailed-description="" \
+    com.docker.extension.publisher-url="" \
+    com.docker.extension.additional-urls="" \
+    com.docker.extension.changelog=""
+
+COPY metadata.json .
+COPY docker.svg .
+COPY --from=client-builder /ui/build ui
+
+```
+
+  </div>
+  <div id="vue-dockerfile" class="tab-pane fade" markdown="1">
+
+<br/>
+
+> **Important**
+>
+> We don't have a working Dockerfile for Vue yet. [Fill out the form](https://docs.google.com/forms/d/e/1FAIpQLSdxJDGFJl5oJ06rG7uqtw1rsSBZpUhv_s9HHtw80cytkh2X-Q/viewform?usp=pp_url&entry.1333218187=Vue)
+> and let us know you'd like a Dockerfile for Vue.
+{: .important }
+
+  </div>
+  <div id="angular-dockerfile" class="tab-pane fade" markdown="1">
+
+<br/>
+
+> **Important**
+>
+> We don't have a working Dockerfile for Angular yet. [Fill out the form](https://docs.google.com/forms/d/e/1FAIpQLSdxJDGFJl5oJ06rG7uqtw1rsSBZpUhv_s9HHtw80cytkh2X-Q/viewform?usp=pp_url&entry.1333218187=Angular)
+> and let us know you'd like a Dockerfile for Angular.
+{: .important }
+
+  </div>
+  <div id="svelte-dockerfile" class="tab-pane fade" markdown="1">
+
+<br/>
+
+> **Important**
+>
+> We don't have a working Dockerfile for Svelte yet. [Fill out the form](https://docs.google.com/forms/d/e/1FAIpQLSdxJDGFJl5oJ06rG7uqtw1rsSBZpUhv_s9HHtw80cytkh2X-Q/viewform?usp=pp_url&entry.1333218187=Svelte)
+> and let us know you'd like a Dockerfile for Svelte.
+{: .important }
+
+  </div>
+</div>
+
+## Configure the metadata file
+
+A `metadata.json` file is required at the root of your extension directory.
+
+```json
+{
+  "icon": "docker.svg",
+  "ui": {
+    "dashboard-tab": {
+      "title": "UI Extension",
+      "root": "/ui",
+      "src": "index.html"
```

Comment:
Could you mention what these 2 props are for? We need to emphasize that they point to the extension's image filesystem (not the host filesystem).